Refined Tests

fbritoferreira opened the pull request:
@brainflake Refined the test to include typesafe in them also realized that some of the actions on the old test code wasn't actually being executed, this should improve the results and have a more indept analysis when testing.

Commit:
Disabled check for `Expected an assignment or function call and instead saw an expression.` on aslant.

Commit:
Increased timeout on function for mocha testing.

Commit:
Refined companies test to have type based checking to data.

Commit:
Refined contacts test to have type based checking to data.

Commit:
Refined deals test to have type based checking to data.

Commit:
Refined email and email events test to have type based checking to data.

Commit:
Refined social media, pipelines and lists test to have type based checking to data.

Commit:
Refined lists test to have type based checking to data.

Commit:
Exposed self variable on the client so it can be used for testing in for the key and token functions.

Commit:
More indent test into function, added missing test for engagements and companies

brainflake commented on lib/client.js:
```diff
@@ -11,15 +11,14 @@ function Client(opts) {
 
-  function useToken (token) {
+  function useToken (token, cb) {
     if (!token || typeof token !== 'string') {
```
I'm not convinced we should be accepting a callback in these. I do see the benefit of consistent error handling, but it seems unnecessary to add an extra parameter for that since these methods aren't doing anything asynchronous. Maybe at the very least we could have a check for `cb` before executing it in the error block? Although I'm thinking just nix the check altogether.

fbritoferreira replied:
That makes sense will remove all callbacks from this function.

brainflake replied:
Actually, I didn't realize we were using `cb` to return an error if a key or token isn't passed in (there were some bugs reported on it awhile back). We can leave it like this for now and maybe clean it up later. Thanks!

brainflake commented on test/contacts.js:
```diff
@@ -10,228 +10,231 @@ client.useKey(api_key);
+      },function(err, data, res) {
+        if (err) { throw err; }
+        expect(res.statusCode).to.equal(404);
+        expect(data.status).to.equal('error');
```
Are we really expecting an error here? If so we should probably rename the test or pull it our altogether.

fbritoferreira replied:
Not sure how this slipped my view but will change it to actually have usefull errors.

Commit:
Removed unnecessary callback in useToken.
Added `client.contacts.createOrUpdate(email, data, cb)` to readme Closes #35

Fixed contacts test.
Small fixes in tests.
